core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -76,16 +86,39 @@ public void insertRole(RoleEntity roleEntity, boolean overwritten) {
           MetalakeMetaService.getInstance().getMetalakeIdByName(roleEntity.namespace().level(0));
       RolePO.Builder builder = RolePO.builder().withMetalakeId(metalakeId);
       RolePO rolePO = POConverters.initializeRolePOWithVersion(roleEntity, builder);
-
-      SessionUtils.doWithCommit(
-          RoleMetaMapper.class,
-          mapper -> {
-            if (overwritten) {
-              mapper.insertRoleMetaOnDuplicateKeyUpdate(rolePO);
-            } else {
-              mapper.insertRoleMeta(rolePO);
-            }
-          });
+      List<SecurableObjectPO> securableObjectPOs = Lists.newArrayList();
+      for (SecurableObject object : roleEntity.securableObjects()) {
+        SecurableObjectPO.Builder objectBuilder =
+            POConverters.initializeSecurablePOBuilderWithVersion(
+                roleEntity.id(), object, getEntityType(object));
+        objectBuilder.withEntityId(
+            MetadataObjectUtils.getSecurableObjectEntityId(
+                metalakeId, object.fullName(), object.type()));
```

Comment:
What's the `fullname` used for?  can we store the all pareants id at the same time, I notice you will split the `fullanme` every time and fetch the corresponding ID from DB.

Reply:
It's similar to NameIdenfitier. The behaviour is similar to other nameidentier's implement. It's inconvenient to store parentId. The length of parents isn't fixed.